spec/java/util/HashSet.Spliterator.lsl

```diff
@@ -0,0 +1,221 @@
+///#! pragma: non-synthesizable
+libsl "1.1.0";
+
+library std
+    version "11"
+    language "Java"
+    url "https://github.com/openjdk/jdk11/blob/master/src/java.base/share/classes/java/util/HashSet.java";
+
+// imports
+
+import java/util/HashSet;
+
+
+// automata
+
+automaton HashSet_KeySpliteratorAutomaton
+(
+    var keysStorage: list<Object>,
+    var index: int,
+    var fence: int,
+    var est: int,
+    var expectedModCount: int,
+    var parent: HashSet
+)
+: HashSet_KeySpliterator
+{
+    // states and shifts
+
+    initstate Allocated;
+    state Initialized;
+
+    shift Allocated -> Initialized by [
+        // constructors
+        HashSet_KeySpliterator
+    ];
+
+    shift Initialized -> self by [
+        // read operations
+        characteristics,
+        trySplit,
+        estimateSize,
+
+        // write operations
+        forEachRemaining,
+        tryAdvance,
+    ];
+
+
+    // utilities
+
+    proc _getFence(): int
+    {
+        action ASSUME(this.parent != null);
+
+        var hi: int = this.fence;
+        if (hi < 0)
+        {
+            val parentStorage: map<Object, Object> = HashSetAutomaton(this.parent).storage;
+            this.est = HashSetAutomaton(this.parent).length;
+            this.expectedModCount = HashSetAutomaton(this.parent).modCount;
+            this.fence = this.est;
+            // That's right ?
+            // Original code: "hi = fence = (tab == null) ? 0 : tab.length;"
+            hi = this.fence;
+        }
+        result = hi;
+    }
+
+
+    @AutoInline proc _throwNPE (): void
+    {
+        action THROW_NEW("java.lang.NullPointerException", []);
+    }
+
+
+    proc _checkForComodification (): void
+    {
+        val modCount: int = HashSetAutomaton(this.parent).modCount;
+        if (this.expectedModCount != modCount)
+            action THROW_NEW("java.util.ConcurrentModificationException", []);
+    }
+
+
+    // constructors
+
+    constructor *.HashSet_KeySpliterator (@target self: HashSet_KeySpliterator, source: HashMap, origin: int, fence: int, est: int, expectedModCount: int)
+    {
+        this.index = origin;
+        this.fence = fence;
+        this.est = est;
+        this.expectedModCount = expectedModCount;
+    }
+
+
+    // methods
+
+    fun *.estimateSize (@target self: HashSet_KeySpliterator): long
+    {
+        _getFence();
+        result = this.est as long;
+    }
+
+
+    fun *.characteristics (@target self: HashSet_KeySpliterator): int
+    {
+        action ASSUME(this.parent != null);
+
+        var mask: int = 0;
+        val length: int = HashSetAutomaton(this.parent).length;
+        if (this.fence < 0 || this.est == length)
+            mask = 64;
```

Comment:
What is this *magic* number?
(hinting on `SPLITERATOR_*` constants declared in `java/util/_interfaces.lsl`)

Reply:
Fixed